### PR TITLE
Try to find home dir, else fail

### PR DESCRIPTION
**Pull request by bf4**
Closes https://github.com/skylightio/skylight-ruby/issues/61

I didn't add any tests at this time, since we might want to first discuss implementation.

### CODE REVIEW

File: lib/skylight/user_config.rb [hunk 1]
```diff
@@ -14,7 +14,18 @@ def initialize
...
+          require "etc"
+          home_dir = File.expand_path("~") || Etc.getpwuid.dir || (ENV["USER"] && File.expand_path("~#{ENV['USER']}"))
+          if home_dir
+            File.join(home_dir, ".skylight")
+          else
+            ""
+          end
+        end
+      # Must return something that is compatible with `File.exist?`
+      config_path.empty? ? config_path || File.expand_path(config_path)
```
Comment (wagenet) [inline lib/skylight/user_config.rb]: Syntax error, should be `:` not `||`.

I'd be ok with raising here if we don't have a path. We can then tell people to set `SKYLIGHT_USER_CONFIG_PATH`.

File: spec/unit/user_config_spec.rb [hunk 2]
```diff
@@ -12,11 +12,30 @@ def set_file_path(path)
...
+        yield
+      ensure
+        ENV.replace(original)
+      end
+    end
+
+    it "preferably uses SKYLIGHT_USER_CONFIG_PATH" do
+      dir = File.expand_path(Dir.mktmpdir)
+      config_path = File.join(dir, 'skylight_user_config_path')
+      FileUtils.touch(config_path)
```
Comment (bf4) [inline spec/unit/user_config_spec.rb]: I didn't realize at first that the config path was actually a path to the config file, not a path to a folder used for config :(.  I could make that more clear in the example config_path variable name and value.

File: lib/skylight/user_config.rb (hunk 1 above)
Comment (bf4) [inline lib/skylight/user_config.rb]: silly github, this is already fixed.

File: lib/skylight/user_config.rb [hunk 3]
```diff
@@ -14,7 +14,19 @@ def initialize
     end
 
     def file_path
-      File.expand_path(ENV["SKYLIGHT_USER_CONFIG_PATH"] || "~/.skylight")
+      return @file_path if @file_path
```
Comment (bf4) [inline lib/skylight/user_config.rb]: @wagenet better IMHO to `return @file_path if defined?(@file_path)` else you'll get a Ruby warning for an uninitialized ivar

**Pull request merged**